core/rlp_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2019 The go-ethereum Authors
```

Comment:
This file seems super out of place. RLP iteration was already supposedly tested in package `rlp`. What does this add in addition? Also if it's just a sanity check that it works for blocks, shouldn't the entire thing be moved into `types`?

Reply:
So, why it doesn't sit in rlp is that it's testing that it works for `*types.Block`, as you guessed. And no, rlp iteration is new, it has no tests in the rlp package. 

I can move it to types...

Reply:
Or rather, moving it to types is a bit difficult too, since I use GenerateChain and a lot of stuff from the core package...

Reply:
See the comment on https://github.com/ethereum/go-ethereum/pull/20302/files#diff-16bfc2f96b958e952d2f3a0e32b8d38cR25 : 

```

// TestIterator tests some basic things about the ListIterator. A more
// comprehensive test can be found in core/rlp_test.go, where we can
// use both types and rlp without dependency cycles
```